build.ps1

```diff
@@ -1,20 +1,79 @@
+param(
+    [string]$targetFramework = 'ALL'
+)
 
 # lets calulat the correct version here
 $fallbackVersion = "1.0.0";
 $version = ''
 
-$tagRegex = '^v?(\d+\.\d+\.\d+)(-([a-zA-Z]+)\.?(\d*))?$'
+$tagRegex = '^v?(\d+\.\d+\.\d+)(?:-([a-zA-Z]+)\.?(\d*))?$'
```

Comment:
https://github.com/marketplace/actions/use-actions

Reply:
we used to use gitversion, it ended up breaking our release numbers so we stopped using it and created our own system for calculating the version numbers.

Reply:
Cool. Didn't know that. I've been using it at work for a while now without issues so was just wondering

Reply:
Got them working well.